mediator/med_methods_mod.F90

```diff
@@ -2532,9 +2532,10 @@ subroutine med_methods_FB_check_for_nans(FB, rc)
 
 #ifndef CESMCOUPLED
     ! For now only CESM uses shr_infnan_isnan - so until other models provide this
-    RETURN
+    mediator_checkfornans = .false.
 #endif
-
+    if(.not. mediator_checkfornans) return
+    
```

Comment:
With the change in how you set the config variable, we can remove the CESMCOUPLED ifdef here.

Reply:
The reason I haven't done that yet is that the github ext test doesn't include the cdeps library and thus the share code.  I am working on updating the github test to include that and then will remove the ifdefs once I have that working.  I got pulled into another issue though so it may be a day or two before I can get back to this, for now I'll convert this PR to draft.

Reply:
OK, thanks. I was worried I was holding you up by not testing this yet.